File: 02-frontend/src/app/model/policypart.ts
import {Policy} from "./policy";
import {Invoice} from "./invoice";

export class PolicyPart {
  public id: number;
  public policy: Policy;
  public invoice: Invoice;
  public part: number;
  public sum: number;
  public date: Date;
  public datePaid: Date;
  public dateConfirmed: Date;
  public sumReal: number;
  public reminder: boolean;

  constructor() {
    this.id = 0;
    this.policy = null;
    this.invoice = null;
    this.part = 0;
    this.sum = 0;
    this.date = null;
    this.datePaid = null;
    this.dateConfirmed = null;
    this.sumReal = null;
    this.reminder = true;
  }

  static fromJson(json: any): PolicyPart {
    const policyPart = new PolicyPart();

    policyPart.id = json.id;
    policyPart.policy = Policy.fromJson(json.policy);
    if(json.invoice) policyPart.invoice = Invoice.fromJson(json.invoice);
    policyPart.part = json.part;
    policyPart.sum = json.sum;

    if (json.date ) policyPart.date = new Date(json.date);
    if (json.datePaid) policyPart.datePaid = new Date(json.datePaid);
    if (json.dateConfirmed) policyPart.dateConfirmed = new Date(json.dateConfirmed);

    policyPart.sumReal = json.sumReal;
    policyPart.reminder = json.reminder;

    return policyPart;
  }

  static fromJsonList(jsonList: any[]): PolicyPart[] {
    return jsonList.map(json => PolicyPart.fromJson(json));
  }
}

File: 01-backend/src/main/java/ee/insa/crmapp/model/PolicyPart.java
package ee.insa.crmapp.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name = "policy_part")
public class PolicyPart {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;
    @OneToOne
    private Policy policy;
    @OneToOne
    private Invoice invoice;
    private Integer part;
    private Double sum;
    private LocalDate date;
    private LocalDate datePaid;
    private LocalDate dateConfirmed;
    private Double sumReal;
    private boolean reminder;
}